sqlair.go

```diff
@@ -25,7 +28,14 @@ var ErrNoRows = sql.ErrNoRows
 // Statement represents a SQL statement with valid SQLair expressions.
 // It is ready to be run on a SQLair DB.
 type Statement struct {
+	id uuid.UUID
 	pe *expr.PreparedExpr
+
+	// A list of *sql.Stmt caches that this Statement appears in. This is
+	// needed to remove and close these the db-prepared statements when the
+	// finalizer for this Statement is run.
+	caches      []*cache
```

Comment:
That creates the same problem that was solved by the ID system above: caches will pile up here until the system runs out of memory. To allow for a single finalizer to exist, we need a double-sided cache so that we can clean both sides when either of them go out of memory. Vaguely:
```
var cacheMutex sync.Mutex
var cacheStmtDB = map[stmtID] = map[dbID] = dbSTMT
var cacheDBStmt = map[dbID] = map[stmtID] = dbSTMT
```

Reply:
Changed.